### PR TITLE
New template for NGFW FortiGate

### PR DESCRIPTION
**Pull request by pnpestov**
The template for processing the command: 
`diagnose lldprx port neighbor details <port-name>`

### CODE REVIEW

File: ntc_templates/templates/fortinet_diagnose_lldprx_port_neighbor_details_port-name.textfsm [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+# Mode		: HA A-P
+# VDOMS     : DISABLE
+#
+# Neighbors	: Huawei CloudEngine S5735-L Series Switches, HPE Aruba 2530 Series Switches, Cisco Catalyst 3000 Series Switches, Cisco Nexus 3000 Series Switches
+#
+# The number of parameters may vary depending on the type of neighbor
+#
+Value PORT (\S+)
+Value PORT_TXT (\S+)
+Value MAC (\S+)
```
Comment (jmcgill298) [inline ntc_templates/templates/fortinet_diagnose_lldprx_port_neighbor_details_port-name.textfsm:17]:
```suggestion
Value MAC_ADDRESS (\S+)

```

Trying to keep these more consistent across templates now

File: ntc_templates/templates/fortinet_diagnose_lldprx_port_neighbor_details_port-name.textfsm [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+Value POWER_STATUS (\S+)
+Value POWER_STATUS_TXT (\S+)
+Value POWER_PAIR (\S+)
+Value POWER_CLASS (\S+)
+Value MAX_FRAME_SIZE (\S+)
+
+Start
+  ^1\s+port:\s+${PORT}
+  ^1\s+port\.txt:\s+${PORT_TXT}
+  ^1\s+mac:\s+${MAC}
```
Comment (jmcgill298) [inline ntc_templates/templates/fortinet_diagnose_lldprx_port_neighbor_details_port-name.textfsm:71]:
```suggestion
  ^1\s+mac:\s+${MAC_ADDRESS}

```

File: ntc_templates/templates/fortinet_diagnose_lldprx_port_neighbor_details_port-name.textfsm [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+  ^1\s+power\.class:\s+${POWER_CLASS}
+  ^1\s+max-frame-size:\s+${MAX_FRAME_SIZE}
+  # Processing possible new information to eliminate errors
+  ^1\s+[\S\s]+
+  ^\s*$$
+  ^. -> Error
+  
+MultipleLines
+  ^${SYSTEM_DESC_DATA}
+  ^1\s+system\.caps\.available:\s+${SYSTEM_CAPS_AVAILABLE} -> Start
```
Comment (jmcgill298) [inline ntc_templates/templates/fortinet_diagnose_lldprx_port_neighbor_details_port-name.textfsm:128]:
```suggestion
  ^1\s+system\.caps\.available:\s+${SYSTEM_CAPS_AVAILABLE} -> Start  
  ^\s*$$
  ^. -> Error
```

File: tests/fortinet/diagnose_lldprx_port_neighbor_details_port-name/fortinet_diagnose_lldprx_port_neighbor_details_port-name.yml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    address_1_type: "1"
+    address_1_type_txt: "ipv4"
+    address_count: "1"
+    aggregation_port: "0"
+    aggregation_status: "1"
+    aggregation_status_txt: "capable"
+    chassis_data: "00:11:22:33:44:55"
+    chassis_type: "4"
+    chassis_type_txt: "interface-mac"
+    mac: "00:11:22:33:44:55"
```
Comment (jmcgill298) [inline tests/fortinet/diagnose_lldprx_port_neighbor_details_port-name/fortinet_diagnose_lldprx_port_neighbor_details_port-name.yml:17]:
```suggestion
    mac_address: "00:11:22:33:44:55"

```

**Comment by pnpestov** [comment]
@jmcgill298 
Thanks for the suggestions! I tried to convey the names of the variables in their original form. But if purposeful work is really underway to systematize the names, then your proposal is rational.

**Comment by pnpestov** [comment]
@jmcgill298 
"CI / yamllint (pull_request) Failing after 1m" - It is suspected that this is because of the character '\r\n' - https://github.com/adrienverge/yamllint/issues/102 ?ysclid=m24s0ytu3361652791 What are my next steps?

**Comment by jmcgill298** [comment]
> @jmcgill298 "CI / yamllint (pull_request) Failing after 1m" - It is suspected that this is because of the character '\r\n' - [adrienverge/yamllint#102](https://github.com/adrienverge/yamllint/issues/102) ?ysclid=m24s0ytu3361652791 What are my next steps?

We have provided scripts to help with this: https://github.com/networktocode/ntc-templates/blob/master/docs/dev/dev_parser.md#development-helper-scripts

**Comment by jmcgill298** [comment]
@pnpestov I created https://github.com/pnpestov/ntc-templates/pull/1

File: ntc_templates/templates/fortinet_diagnose_lldprx_port_neighbor_details_port-name.textfsm [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+# VDOMS     : DISABLE
+#
+# Neighbors	: Huawei CloudEngine S5735-L Series Switches, HPE Aruba 2530 Series Switches, Cisco Catalyst 3000 Series Switches, Cisco Nexus 3000 Series Switches
+#
+# The number of parameters may vary depending on the type of neighbor
+#
+Value PORT (\S+)
+Value PORT_TXT (\S+)
+Value MAC_ADDRESS (\S+)
+
```
Comment (jmcgill298) [inline ntc_templates/templates/fortinet_diagnose_lldprx_port_neighbor_details_port-name.textfsm:18]:
```suggestion

```

File: ntc_templates/templates/fortinet_diagnose_lldprx_port_neighbor_details_port-name.textfsm [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+Value POWER_STATUS_TXT (\S+)
+Value POWER_PAIR (\S+)
+Value POWER_CLASS (\S+)
+Value MAX_FRAME_SIZE (\S+)
+
+Start
+  ^1\s+port:\s+${PORT}
+  ^1\s+port\.txt:\s+${PORT_TXT}
+  ^1\s+mac:\s+${MAC_ADDRESS}
+
```
Comment (jmcgill298) [inline ntc_templates/templates/fortinet_diagnose_lldprx_port_neighbor_details_port-name.textfsm:72]:
```suggestion


```

**Comment by pnpestov** [comment]
@jmcgill298 
Thanks for the help!

**Pull request merged**